### ISSUE TITLE
Python test DiscreteCompoundDistribution_std fails on riscv64

### ISSUE THREAD

**Issue by pgrt** [openturns/openturns#2550]
### What happened?

Hello,

During a rebuild of openturns on the Debian riscv64 autobuilder, DiscreteCompoundDistribution_std failed:

302/594 Test  #837: pyinstallcheck_DiscreteCompoundDistribution_std ........................***Failed    4.17 sec
Traceback (most recent call last):
  File "/<<PKGBUILDDIR>>/python/test/t_DiscreteCompoundDistribution_std.py", line 18, in <module>
    ott.assert_almost_equal(
  File "/<<PKGBUILDDIR>>/debian/tmp/usr/lib/python3/dist-packages/openturns/testing.py", line 174, in assert_almost_equal
    return _testing.assert_almost_equal(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Value 0.00128141 is not close enough to 0.00128136 

The relative difference between these two values is close to 4 E-5 > 1E-5 and the absolute difference is close to 5E-8 > 1E-8, the two rhs being the hardcoded thresholds.

I wonder if there is some computation issue for these values or if one should raise the acceptance thresholds.

### How to reproduce the issue?

```shell
See full build log
https://buildd.debian.org/status/fetch.php?pkg=openturns&arch=riscv64&ver=1.21.1-4%2Bb1&stamp=1708164046&raw=0
```

### Version

1.21.1

### Operating System

Linux

### Installation media

from source

**Comment by jschueller** [comment]
I guess its enough to loosen the tolerances, do this work ?
```
diff --git a/python/test/t_DiscreteCompoundDistribution_std.py b/python/test/t_DiscreteCompoundDistribution_std.py
index a38ece875..9ba90a054 100755
--- a/python/test/t_DiscreteCompoundDistribution_std.py
+++ b/python/test/t_DiscreteCompoundDistribution_std.py
@@ -16,7 +16,7 @@ poisson_distribution = ot.Poisson(10.0)
 
 for i in range(upper_bound):
     ott.assert_almost_equal(
-        distribution.computePDF([i]), poisson_distribution.computePDF([i])
+        distribution.computePDF([i]), poisson_distribution.computePDF([i], 1e-4, 1e-7)
     )
 
 print("Distribution ", repr(distribution))
```

**Comment by pgrt** [comment]
Hi Julien,

Really sorry for answering late, I had trouble accessing a riscv64 machine and it is so slow...

The message I posted occurred for i = 31 in the loop. Then I loosened tolerance as you suggest and it passes for i = 31, unfortunately another error fills in the place for i = 34:

---------------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pgt/openturns-1.21.1/python/test/t_DiscreteCompoundDistribution_std.py", line 18, in <module>
    ott.assert_almost_equal(
  File "/home/pgt/openturns-1.21.1/debian/tmp/usr/lib/python3/dist-packages/openturns/testing.py", line 174, in assert_almost_equal
    return _testing.assert_almost_equal(*args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: Value 3.57974e-05 is not close enough to 3.56884e-05
---------------------------------------------------------------------------------------------------

Which corresponds to a relative error of 3e-3 and an absolute error of 2e-7.

Then for i = 35:
RuntimeError: Value 1.03626e-05 is not close enough to 1.01967e-05

for i = 36:
RuntimeError: Value 3.07107e-06 is not close enough to 2.83241e-06

and so on. The relative errors become quite large. I think we should investigate the computation of the high quantiles there. If you have no clue I will give it a try also.

Thanks for your help,

Pierre

**Comment by pgrt** [comment]
Hi again,

I just saw that a full rebuild of the openturns Debian package has been triggered, and it fails on all architectures (amd64 among others, so) because of the same problem as above.

So I guess something has changed in a dependency recently, as version 1.22 was building successfully before.

**Comment by jschueller** [comment]
oh, I forgot about that one: the swig 4.2 update needed a small fix in this test:
https://github.com/openturns/openturns/commit/ff713c2520d5874b9a8a02cd313b95156bc3ec72
(already part of 1.22)

**Comment by pgrt** [comment]
Thanks a lot, this solved the issue on 1.21.1, as you thought!

Best,
Pierre
